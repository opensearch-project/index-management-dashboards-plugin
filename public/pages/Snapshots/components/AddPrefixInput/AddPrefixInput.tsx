/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { EuiFormRow, EuiFieldText, EuiSpacer } from "@elastic/eui";
import React, { useState, ChangeEvent } from "react";
import CustomLabel from "../../../../components/CustomLabel";

interface AddPrefixesInputProps {
  getPrefix: (prefix: string) => void;
}

const AddPrefixesInput = ({ getPrefix }: AddPrefixesInputProps) => {
  const [prefix, setPrefix] = useState("restored_");

  const onPrefixChange = (e: ChangeEvent<HTMLInputElement>) => {
    setPrefix(e.target.value);
    getPrefix(e.target.value);
  };

  return (
    <>
      <EuiSpacer size="l" />

      <CustomLabel title="Specify prefix for restored index names" helpText="A prefix will be added to any restored index names." />
      <EuiFormRow>
        <EuiFieldText value={prefix} onChange={onPrefixChange} data-test-subj="prefixInput" />
      </EuiFormRow>

      <EuiSpacer size="m" />
    </>
  );
};

export default AddPrefixesInput;
