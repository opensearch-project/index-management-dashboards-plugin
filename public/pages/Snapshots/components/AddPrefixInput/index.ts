/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import AddPrefixInput from "./AddPrefixInput";

export default AddPrefixInput;
