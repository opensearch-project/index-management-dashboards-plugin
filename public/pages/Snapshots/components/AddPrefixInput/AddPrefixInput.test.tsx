/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render, screen, cleanup } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import AddPrefixInput from "./AddPrefixInput";

const testProps = { getPrefix: jest.fn() };

beforeEach(() => {
  render(<AddPrefixInput {...testProps} />);
});

afterEach(() => {
  cleanup();
});

describe("AddPrefixInput component", () => {
  it("renders without error", () => {
    expect(screen.getByText("Specify prefix for restored index names")).toBeInTheDocument();
    expect(screen.getByTestId("prefixInput")).toBeInTheDocument();

    const { container } = render(<AddPrefixInput {...testProps} />);

    expect(container).toMatchSnapshot();
  });

  it("accepts user input", () => {
    userEvent.type(screen.getByTestId("prefixInput"), "test_prefix_");

    expect(screen.getByTestId("prefixInput")).toHaveValue("restored_test_prefix_");
  });

  it("sends user input to parent component via getPrefix", () => {
    userEvent.type(screen.getByTestId("prefixInput"), "four");

    // getPrefix is prop sent from parent component to retrieve user input
    expect(testProps.getPrefix).toBeCalledTimes(4);
  });
});
