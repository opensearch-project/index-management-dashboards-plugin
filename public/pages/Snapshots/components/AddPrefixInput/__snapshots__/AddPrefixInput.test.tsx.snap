// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPrefixInput component renders without error 1`] = `
<div>
  <div
    class="euiSpacer euiSpacer--l"
  />
  <div
    class="euiFlexGroup euiFlexGroup--gutterExtraSmall euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow euiFlexGroup--responsive"
  >
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <div
        class="euiText euiText--small"
      >
        <h3>
          Specify prefix for restored index names
        </h3>
      </div>
    </div>
  </div>
  <span
    style="font-weight: 200; font-size: 12px;"
  >
    A prefix will be added to any restored index names.
  </span>
  <div
    class="euiSpacer euiSpacer--s"
  />
  <div
    class="euiFormRow euiFormRow--compressed"
    id="some_html_id-row"
  >
    <div
      class="euiFormRow__fieldWrapper"
    >
      <div
        class="euiFormControlLayout euiFormControlLayout--compressed"
      >
        <div
          class="euiFormControlLayout__childrenWrapper"
        >
          <input
            class="euiFieldText euiFieldText--compressed"
            data-test-subj="prefixInput"
            id="some_html_id"
            type="text"
            value="restored_"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="euiSpacer euiSpacer--m"
  />
</div>
`;
